### PR TITLE
fix(datahub-frontend): OIDC discovery URL will not have NONE as auth_methods_supported

### PR DESCRIPTION
**Pull request by chen4119**
## Checklist
- [ ] The PR conforms to DataHub's [Contributing Guideline](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [ ] Links to related issues (if applicable)
- [ ] Tests for the changes have been added/updated (if applicable)
- [ ] Docs related to the changes have been added/updated (if applicable). If a new feature has been added a Usage Guide has been added for the same.
- [ ] For any breaking change/potential downtime/deprecation/big changes an entry has been made in [Updating DataHub](https://github.com/datahub-project/datahub/blob/master/docs/how/updating-datahub.md)

### CODE REVIEW

File: datahub-frontend/app/auth/sso/oidc/custom/CustomOidcAuthenticator.java [hunk 1]
```diff
@@ -168,12 +175,4 @@ public void validate(final OidcCredentials credentials, final WebContext context
       }
     }
   }
-
```
Comment (jjoyce0510) [inline datahub-frontend/app/auth/sso/oidc/custom/CustomOidcAuthenticator.java:171]: why remove these? any particular reason?

File: datahub-frontend/app/auth/sso/oidc/custom/CustomOidcAuthenticator.java [hunk 2]
```diff
@@ -132,6 +130,16 @@ private static ClientAuthenticationMethod firstSupportedMethod(final List<Client
     }
   }
 
+  private TokenRequest createTokenRequest(final AuthorizationGrant grant) {
```
Comment (jjoyce0510) [inline datahub-frontend/app/auth/sso/oidc/custom/CustomOidcAuthenticator.java:133]: nice! One small request - we tend to put invoked private methods below public methods as code style. Can you move this method to lower down in the file?

File: datahub-frontend/app/auth/sso/oidc/custom/CustomOidcAuthenticator.java [hunk 3]
```diff
@@ -83,15 +83,13 @@ public CustomOidcAuthenticator(final OidcConfiguration configuration, final Oidc
...
-      clientAuthentication = new EmptyAuthentication(_clientID);
-    } else if (ClientAuthenticationMethod.CLIENT_SECRET_POST.equals(chosenMethod)) {
+    if (ClientAuthenticationMethod.CLIENT_SECRET_POST.equals(chosenMethod)) {
       final Secret _secret = new Secret(configuration.getSecret());
       clientAuthentication = new ClientSecretPost(_clientID, _secret);
     } else if (ClientAuthenticationMethod.CLIENT_SECRET_BASIC.equals(chosenMethod)) {
       final Secret _secret = new Secret(configuration.getSecret());
       clientAuthentication = new ClientSecretBasic(_clientID, _secret);
-    } else {
+    } else if (!ClientAuthenticationMethod.NONE.equals(chosenMethod)) {
```
Comment (jjoyce0510) [inline datahub-frontend/app/auth/sso/oidc/custom/CustomOidcAuthenticator.java:92]: can we explicitly handle none to make reading a bit easier? 

for example, 

else if (ClientAuthenticationMethod.NONE.equals(chosenMethod)) {
   clientAuthentication = null; // No client authentication in none mode. 
}

**Review by jjoyce0510 — COMMENTED** [review]
Overall looks good. Few minor comments then lets merge

**Comment by chen4119** [comment]
Just fixed the AuthorizationGrant compilation error

**Review by jjoyce0510 — APPROVED** [review]
LGTM. Thank you!

**Pull request merged**